src/utils/rlp.cairo

```diff
@@ -140,4 +141,86 @@ namespace RLP {
             return (rlp_len=data_len + 1);
         }
     }
+
+    // @notice encodes felt into an rlp element
+    // @param element The felt that is encoded into rlp
+    // @param rlp_len The length of the rlp array
+    // @param rlp The pointer receiving the rlp encoded felt
+    // @return rlp_len The length of the encoded felt in bytes
+    func encode_element_felt{
+        syscall_ptr: felt*,
+        pedersen_ptr: HashBuiltin*,
+        bitwise_ptr: BitwiseBuiltin*,
+        range_check_ptr,
+    }(element: felt, rlp_len: felt, rlp: felt*) -> (rlp_len: felt) {
+        alloc_locals;
+        // 127 is the largest value that can be represented by one byte
+        let element_is_single_byte = is_le(element, 127);
+
+        if (element_is_single_byte != FALSE) {
+            // single byte needs no prefix
+            assert [rlp + rlp_len] = element;
+            return (rlp_len + 1,);
+        } else {
+            // otherwise, we need to convert the value to an unpadded byte array
+            let (local reversed_element_bytes: felt*) = alloc();
+            let (local element_bytes: felt*) = alloc();
+
+            let (element_high, element_low) = split_felt(element);
+            let (element_bytes_len) = Helpers.uint256_to_bytes_no_padding(
```

Comment:
when testing against the case where the salt was sufficiently large [255,` 2**55-1), the Helpers.felt_to_bytes_array was returning incorrect results.

This motivated creating this helper that encoded a uint256 with no padding for RLP encoding.

Reply:
we have a terrible naming imho from the very beginning regarding `bytes`, `padding` etc.

We should strive to use the same naming convention as the EVM, ie:
- word = `bytes32` = `uint256`
- `bytes_i_` is the _i_ first bytes from the left:
    - `bytes31(bytes32)` will drop the right most byte
    - `bytes1(bytes32)` will keep the left most byte
- `uint_i_` works the opposite (as uint are big endian)
    - for example, with no overflow, `uint64(uint256)` = `uint64`
    - so `uint248(uint256)` will drop the left most byte
    - `uint8(uint256)` will keep the right most byte

`bytes` is a `bytes1[]`, ie an array of bytes and consequently should not have any leading or trailing 0s added for padding

In short, I think that we should once for all create a `BytesLib`, put all these helpers with standard naming conventions and tests, and stop to think about it!

Reply:
Are you ok with `Helpers.uint256_to_bytes_no_padding` as a name for now?

Reply:
can you add an issue for this refactoring?

Reply:
np! see #457